plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -44,14 +44,49 @@ def watch_deployment
     # TODO: hunt down this issue and fix it properly
     Watchers::PodWatcher ; Watchers::ReplicationControllerWatcher ; Watchers::EventWatcher
 
-    # Commented out for now, to avoid creating threads without ever destroying them
-    # Thread.new { kuber_release.watch_pods }
-    # Thread.new { kuber_release.watch_rcs }
+    Thread.new do
+      kuber_release.watch_pods(&method(:handle_pod_notice))
+    end
+
     # Thread.new { sleep(5) ; kuber_release.watch_pod_events }
   end
 
   private
 
+  def handle_pod_notice(notice, watcher)
+    release_doc = kuber_release.docs_by_role[notice.role]
+
+    if release_doc.nil?
+      Kubernetes::Util.log "ERROR: could not find role", role: notice.role, available_roles: kuber_release.docs_by_role.keys
+      return
+    end
+
+    ready_count = watcher.num_ready
+    if ready_count > release_doc.replicas_live
+      release_doc.update_replica_count(ready_count)
+      release_doc.save!
+      update_release_status
+      log 'Another replica is live', role: notice.role, count: ready_count, r_status: kuber_release.status, r_doc_status: release_doc.status
+    end
+  end
+
+  def update_release_status
+    statuses = kuber_release.release_docs.map(&:status)
+
+    if statuses.all? { |s| s == 'live' }
+      deploy_is_live
+    elsif statuses.include?('spinning_up')
+      kuber_release.update_attribute(:status, 'spinning_up') unless kuber_release.status == 'spinning_up'
```

Comment:
Should we be asking the`kuber_release` if it's on a `spinning up` state instead of making this comparison, i.e., `kuber_release.spinning_up?`.

If we need to make this same comparison somewhere else, it would be nice to have this logic centralised in the `Kubernetes::Release` object.